addons/delivery/models/delivery_carrier.py

```diff
@@ -47,10 +33,39 @@ class DeliveryCarrier(models.Model):
     # Internals for shipping providers #
     # -------------------------------- #
 
-    delivery_type = fields.Selection([('grid', 'Default Delivery')], string="Delivery Type", default='grid', required='True')
-
+    sequence = fields.Integer(help="Determine the display order", default=10)
+    delivery_type = fields.Selection([('fixed', 'Fixed Price'), ('base_on_rule', 'Based on Rules')], string='Price Computation', default='fixed', required=True)
+    product_type = fields.Selection(related='product_id.type', default='service')
+    product_sale_ok = fields.Boolean(related='product_id.sale_ok', default=False)
+    partner_id = fields.Many2one('res.partner', string='Transporter Company', required=True, help="The partner that is doing the delivery service.")
+    product_id = fields.Many2one('product.product', string='Delivery Product', required=True, ondelete="cascade")
```

Comment:
set null is the default value. ;) so we should keep ondelete='cascade' if we want to remove child reference when parent is removed.